File: Examples/PackageDiagram/TypeScript/Banking/TransferService.ts
import { Account } from "../Core/Account";
import { AccountRepository } from "../Core/AccountRepository";
import { Money } from "../Core/Money";

/** Moves money between accounts. Part of the `Banking` module; depends on `Core`. */
export class TransferService {
    constructor(private readonly repository: AccountRepository) {}

    transfer(amount: Money, source: string, destination: string): void {
        const sender = this.repository.find(source);
        const recipient = this.repository.find(destination);
        if (!sender || !recipient) {
            return;
        }
        sender.deposit(amount);
        recipient.deposit(amount);
        this.repository.save(sender);
        this.repository.save(recipient);
    }
}

File: Examples/PackageDiagram/Python/Banking/transfer_service.py
from account_repository import AccountRepository
from money import Money


class TransferService:
    """Moves money between accounts. Part of the ``Banking`` module; depends on ``Core``."""

    def __init__(self, repository: AccountRepository):
        self.repository: AccountRepository = repository

    def transfer(self, amount: Money, source: str, destination: str) -> None:
        sender = self.repository.find(source)
        recipient = self.repository.find(destination)
        if sender is None or recipient is None:
            return
        sender.deposit(amount)
        recipient.deposit(amount)
        self.repository.save(sender)
        self.repository.save(recipient)
